File: submitted_code/100298.fan-zhuan-lian-biao-lcof.cpp
//
// @lc app=leetcode.cn id=100298 lang=cpp
//
// [100298] fan-zhuan-lian-biao-lcof
//
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if(!head)
            return NULL;
        ListNode *p = NULL;
        ListNode *he = head;
        ListNode *ne = he->next;
        while(ne){
            he->next = p;
            p = he;
            he = ne;
            ne = ne->next;
        }
        he->next = p;
        return he;
    }
};
// @lc code=end

File: submitted code/100298.fan-zhuan-lian-biao-lcof.java
//
// @lc app=leetcode.cn id=100298 lang=java
//
// [100298] fan-zhuan-lian-biao-lcof
//
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode reverseList(ListNode head) {
        if(head == null || head.next == null)
            return head;
        ListNode p = head, q = head.next;
        p.next = null;
        while(q != null){
            ListNode t = q.next;
            q.next = p;
            p = q;
            q = t;
        }
        return p;
    }
}
// @lc code=end